cli/internal/devserver/filesync.go

```diff
@@ -65,11 +67,39 @@ func (fs *NitricFileSync) setFileContents(contents []byte) error {
 	return err
 }
 
+func (fs *NitricFileSync) checkFileValidity(contents []byte) ([]schema.ValidationError, error) {
```

Comment:
I think this name could be a little better, to make it clear what file it's referring to. E.g. `validateAppSchemaYaml` or something like that

Reply:
It also look like this doesn't need to be attached to the struct (i.e. remove `(fs *NitricFileSync)`)